homeassistant/components/backup/http.py

```diff
@@ -42,12 +42,13 @@ async def get(
 
         manager = cast(BackupManager, request.app[KEY_HASS].data[DATA_MANAGER])
         backup = await manager.async_get_backup(slug=slug)
+        path = await manager.async_get_backup_path(slug=slug)
 
-        if backup is None or not backup.path.exists():
+        if backup is None or path is None or not path.exists():
```

Comment:
We need to run the exists check in the executor.

Reply:
I'd replace this check with try... except FileNotFoundError when reading the file though since the file can be removed at any time.

Reply:
It seems aiohttp already checks that, we don't need to do it